### PR TITLE
Add pan events

### PR DESCRIPTION
**Pull request by szbergeron**
WIP: plumbs support for smooth scrolling

Adds a new "MousePan" event type that matches up with https://github.com/revery-ui/esy-sdl2/pull/6

Needs to be integrated with Revery proper before it will work

### CODE REVIEW

**Comment by szbergeron** [comment]
Builds probably failing due to missing symbols (relies on changes in esy-sdl2#6)

File: src/sdl2_wrapper.cpp [hunk 1]
```diff
@@ -401,6 +401,8 @@ CAMLprim value Val_SDL_Event(SDL_Event *event) {
 
   int tag, mouseButton;
 
+  printf("event type is %d while pan event type is %d and mousewheel is %d\n", event->type, SDL_PANEVENT, SDL_MOUSEWHEEL);
```
Comment (bryphe) [inline src/sdl2_wrapper.cpp]: We should remove the `printf`s for the final version

File: src/sdl2_wrapper.cpp [hunk 2]
```diff
@@ -551,6 +567,7 @@ CAMLprim value Val_SDL_Event(SDL_Event *event) {
       v = Val_SDL_WindowEvent(23, event->window.windowID);
       break;
     default:
+      printf("Unknown event in sdl2_wrapper, event enum code was %d\n", event->type);
```
Comment (Et7f3) [inline src/sdl2_wrapper.cpp]:
```suggestion
      fprintf(stderr, "Unknown event in sdl2_wrapper, event enum code was %d\n", event->type);
```

**Review by Et7f3 — APPROVED** [review]

**Comment by bryphe** [comment]
I think if you bring in the latest master, this should be build now @szbergeron 👍

File: src/sdl2_wrapper.cpp [hunk 3]
```diff
@@ -551,6 +567,7 @@ CAMLprim value Val_SDL_Event(SDL_Event *event) {
       v = Val_SDL_WindowEvent(23, event->window.windowID);
       break;
     default:
+      fprintf(stderr, "Unknown event in sdl2_wrapper, event enum code was %d\n", event->type);
```
Comment (bryphe) [inline src/sdl2_wrapper.cpp]: We should remove this as well - might cause issues in the case where we run on Windows w/o a console allocated (we could hook into the `SDL_Log` mechanism though instead)

**Review by bryphe — CHANGES_REQUESTED** [review]
One minor change - we should remove the `fprintf`. Otherwise, everything else looks good - it should build now w/ `master` merged

**Comment by bryphe** [comment]
Looks great, @szbergeron ! Thank you!

**Pull request merged**